app/routes.py

```diff
@@ -1,2 +1,239 @@
-from flask import Blueprint
+from flask import Blueprint, json, jsonify, request, make_response
+from werkzeug.wrappers import response
+from app.models.task import Task
+from app.models.goal import Goal
+from app import db
+import requests
+from datetime import datetime
+import os
+from dotenv import load_dotenv
+
+tasks_bp = Blueprint("tasks", __name__, url_prefix="/tasks")
+
+@tasks_bp.route("", methods=["POST"])
+def post_tasks():
+    request_body = request.get_json()
+    if "title" not in request_body or "description" not in request_body or "completed_at" not in request_body:
+        return make_response({"details":"Invalid data"}, 400)
+    
+    new_task = Task(
+        title = request_body["title"],
+        description = request_body["description"],
+        completed_at = request_body["completed_at"]
+        )
+    db.session.add(new_task)
+    db.session.commit()
+    
+    response_body = {"task": {"id": new_task.id,
+        "title": new_task.title,
+        "description": new_task.description,
+        "is_complete": bool(new_task.completed_at)}}
+        
+    return jsonify(response_body), 201
+
+
+@tasks_bp.route("", methods=["GET"])   
+def get_tasks():
+        sort_query = request.args.get("sort")
+        if sort_query == "asc":
+            tasks = Task.query.order_by(Task.title.asc()).all()
+        elif sort_query == "desc":
+            tasks = Task.query.order_by(Task.title.desc()).all()
+        else:
+            tasks = Task.query.all()
+
+        tasks_response = []
+        for task in tasks: 
+            tasks_response.append({
+                "id": task.id,
+                "title": task.title,
+                "description": task.description,
+                "is_complete": bool(task.completed_at)
+                })
+
+        return jsonify(tasks_response), 200
+
+
+@tasks_bp.route("/<task_id>", methods=["GET", "PUT", "DELETE"])
+def handle_task(task_id):
+    task = Task.query.get_or_404(task_id)
+    if not task:
+        return make_response(f"Task {task_id} not found", 404)
+        
+    if request.method == "GET":
+        if not task.goal_id:
+            response_body = {"task":
+                {"id": task.id,
+                "title": task.title,
+                "description": task.description,
+                "is_complete": bool(task.completed_at)}}
+            return jsonify(response_body), 200
+        
+        else:
+            response_body = {"task":
+            {"id": task.id,
+            "goal_id": task.goal_id,
+            "title": task.title,
+            "description": task.description,
+            "is_complete": bool(task.completed_at)}}
+            return jsonify(response_body), 200
+
+    elif request.method == "PUT":
+        request_body = request.get_json()
+        task.title = request_body['title']
+        task.description = request_body['description']
```

Comment:
We should be sure that the same fields required for POSTing are included here for PUT. PUT _replaces_ the value for the supplied task id, so we should ensure that all of the values required to represent a `Task` are provided with the same restrictions as we had when creating it in the first place.

At the least, we should ensure that the keys are present in the dictionary before trying to access them, otherwise this request would cause a `KeyError`.